bdk_chain/src/file_store.rs

```diff
@@ -107,22 +129,30 @@ where
         &mut self,
         changeset: &KeychainChangeSet<K, P>,
     ) -> Result<(), io::Error> {
-        if !changeset.is_empty() {
```

Comment:
^ might wanna expand documentation to explain that this function will always set the length of the file to end after the changeset you added.